aws/resource_aws_route53_record.go

```diff
@@ -754,35 +745,25 @@ func resourceAwsRoute53RecordDelete(d *schema.ResourceData, meta interface{}) er
 }
 
 func deleteRoute53RecordSet(conn *route53.Route53, input *route53.ChangeResourceRecordSetsInput) (interface{}, error) {
-	wait := resource.StateChangeConf{
-		Pending:    []string{"rejected"},
-		Target:     []string{"accepted"},
-		Timeout:    5 * time.Minute,
-		MinTimeout: 1 * time.Second,
-		Refresh: func() (interface{}, string, error) {
-			resp, err := conn.ChangeResourceRecordSets(input)
-			if err != nil {
-				if r53err, ok := err.(awserr.Error); ok {
-					if r53err.Code() == "PriorRequestNotComplete" {
-						// There is some pending operation, so just retry
-						// in a bit.
-						return 42, "rejected", nil
-					}
-
-					if r53err.Code() == "InvalidChangeBatch" {
-						// This means that the record is already gone.
-						return resp, "accepted", nil
-					}
-				}
-
-				return 42, "failure", err
-			}
-
-			return resp, "accepted", nil
-		},
+	var out *route53.ChangeResourceRecordSetsOutput
+	err := resource.Retry(1*time.Minute, func() *resource.RetryError {
+		var err error
+		out, err = conn.ChangeResourceRecordSets(input)
+		if isAWSErr(err, "InvalidChangeBatch", "") {
+			// This means the record is already gone
+			return nil
+		}
+		return resource.NonRetryableError(err)
+	})
+	if isResourceTimeoutError(err) {
+		out, err = conn.ChangeResourceRecordSets(input)
+		if isAWSErr(err, "InvalidChangeBatch", "") {
+			// This means the record is already gone
+			return out, nil
+		}
```

Comment:
This can be simplified without `resource.Retry()` at all since there are no retryable conditions 👍 

```go
out, err := conn.ChangeResourceRecordSets(input)

if isAWSErr(err, "InvalidChangeBatch", "") {
  return out, nil
}

return out, err
```

See also: https://github.com/terraform-providers/terraform-provider-aws/issues/11864

Reply:
@bflad 

+1

fixed with https://github.com/terraform-providers/terraform-provider-aws/pull/11895/commits/f5cf401f5cec685033d519c03233bdce3f112173